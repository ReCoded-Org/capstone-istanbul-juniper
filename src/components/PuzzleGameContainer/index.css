.gameContainer {
  background-color: #3b8c79;
  width: 70%;
  height: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 3%;
}

.gameContainer .ant-card-body {
  display: flex;
  flex-direction: row;
}

.gameContainer__listOfImages {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 20%;
}

.gameContainer__puzzleGrid {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
}

.gameContainer__listOfImages--imgContainer {
  height: auto;
  transition: transform 0.4s;
  width: 100%;
}

.gameContainer__listOfImages--img {
  object-fit: contain;
  max-width: 100%;
  max-height: auto;
}

.gameContainer__listOfImages--imgContainer:hover {
  cursor: pointer;
  transform: scale(1.3);
}

@media (max-width: 425px) {
  .gameContainer.ant-card {
    width: 100%;
  }

  .gameContainer .ant-card-body {
    flex-direction: column;
  }

  .gameContainer__listOfImages {
    flex-direction: row;
    width: 100%;
  }
}
