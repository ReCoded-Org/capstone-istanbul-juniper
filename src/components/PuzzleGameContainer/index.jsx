import React, { useState } from "react";
import { Col, Row, Card } from "antd";
import habitatImg from "./../../images/puzzleGame/habitat.jpg";
import conservationImg from "./../../images/puzzleGame/conservation.jpg";
import holdingPlantImg from "./../../images/puzzleGame/holdingPlant.jpg";
import plantingImg from "./../../images/puzzleGame/planting.jpg";
import PuzzleGrid from "./../PuzzleGrid";
import "./index.css";

const PUZZLE_IMAGE_POOL = [
  habitatImg,
  conservationImg,
  holdingPlantImg,
  plantingImg,
];

// This component contains the list of images the user can choose to solve
// it also renders the PuzzleGrid component
// When the user clicks on an image it will be passed to the PuzzleGrid component as a prop
// PuzzleGrid then renders the puzzle tiles of that image
const PuzzleGameContainer = () => {
  // currentImage is the image displayed for the puzzle.
  // It will be divided into pieces using CSS.
  const [currentImage, setCurrentImage] = useState(PUZZLE_IMAGE_POOL[0]);

  const handleImageClick = (e) => {
    setCurrentImage(e.target.getAttribute("src"));
  };

  return (
    <Card size="small" className="gameContainer">
      <Row className="gameContainer__listOfImages">
        {PUZZLE_IMAGE_POOL.map((img) => {
          return (
            <Col
              onClick={(e) => handleImageClick(e)}
              className="gameContainer__listOfImages--imgContainer"
            >
              <img
                className="gameContainer__listOfImages--img"
                src={img}
                alt="Click to choose your puzzle"
              />
            </Col>
          );
        })}
      </Row>
      <Row className="gameContainer__puzzleGrid">
        <PuzzleGrid puzzleImage={currentImage} />
      </Row>
    </Card>
  );
};

export default PuzzleGameContainer;
