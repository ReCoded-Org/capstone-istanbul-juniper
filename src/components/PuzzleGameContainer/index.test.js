import React from "react";
import PuzzleGameContainer from "./index.jsx";
import renderer from "react-test-renderer";

describe("PuzzleGameContainer Component", () => {
  it("matches the tree", () => {
    const tree = renderer.create(<PuzzleGameContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
