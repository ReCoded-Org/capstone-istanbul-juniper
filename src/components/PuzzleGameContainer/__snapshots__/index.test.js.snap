// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PuzzleGameContainer Component matches the tree 1`] = `
<div
  className="ant-card ant-card-bordered ant-card-small gameContainer"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-row gameContainer__listOfImages"
      style={Object {}}
    >
      <div
        className="ant-col gameContainer__listOfImages--imgContainer"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="Click to choose your puzzle"
          className="gameContainer__listOfImages--img"
          src="habitat.jpg"
        />
      </div>
      <div
        className="ant-col gameContainer__listOfImages--imgContainer"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="Click to choose your puzzle"
          className="gameContainer__listOfImages--img"
          src="conservation.jpg"
        />
      </div>
      <div
        className="ant-col gameContainer__listOfImages--imgContainer"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="Click to choose your puzzle"
          className="gameContainer__listOfImages--img"
          src="holdingPlant.jpg"
        />
      </div>
      <div
        className="ant-col gameContainer__listOfImages--imgContainer"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="Click to choose your puzzle"
          className="gameContainer__listOfImages--img"
          src="planting.jpg"
        />
      </div>
    </div>
    <div
      className="ant-row gameContainer__puzzleGrid"
      style={Object {}}
    >
      <div
        className="ant-col puzzleGrid"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
